src/main/java/teammates/sqllogic/core/UsersLogic.java

```diff
@@ -205,6 +212,74 @@ public List<User> getAllUsersByGoogleId(String googleId) {
         return usersDb.getAllUsersByGoogleId(googleId);
     }
 
+    /**
+     * Common requirement between reset user by googleId function.
+     * Refer to {@link #resetInstructorGoogleId(String, String, String)} or
+     * {@link #resetStudentGoogleId(String, String, String)}.
+     */
+    private void deleteAccountCascadeAll(String googleId) {
+        if (usersDb.getAllInstructorsByGoogleId(googleId).isEmpty()
+                && usersDb.getAllStudentsByGoogleId(googleId).isEmpty()) {
+            accountsLogic.deleteAccountCascade(googleId);
+        }
+    }
+
+    /**
+     * Resets the googleId associated with the instructor.
+     */
+    public void resetInstructorGoogleId(String email, String courseId, String googleId)
+            throws EntityDoesNotExistException {
+        assert email != null;
+        assert courseId != null;
+        assert googleId != null;
+
+        Instructor instructor = getInstructorForEmail(courseId, email);
+
+        if (instructor == null) {
+            // Is there a better message?
+            throw new EntityDoesNotExistException(ERROR_UPDATE_NON_EXISTENT
+                    + "Instructor [courseId=" + courseId + ", email=" + email + "]");
+        }
+
+        instructor.setAccount(null);
+
+        List<Account> accounts = accountsLogic.getAccountsForEmail(email);
+
+        for (Account account : accounts) {
+            accountsLogic.deleteAccount(account.getGoogleId());
+        }
+
+        deleteAccountCascadeAll(googleId);
+    }
+
+    /**
+     * Resets the googleId associated with the student.
+     */
+    public void resetStudentGoogleId(String email, String courseId, String googleId)
+            throws EntityDoesNotExistException {
+        assert email != null;
+        assert courseId != null;
+        assert googleId != null;
+
+        Student student = getStudentForEmail(courseId, email);
+
+        if (student == null) {
+            // Is there a better message?
+            throw new EntityDoesNotExistException(ERROR_UPDATE_NON_EXISTENT
+                    + "Student [courseId=" + courseId + ", email=" + email + "]");
+        }
+
+        student.setAccount(null);
+
+        List<Account> accounts = accountsLogic.getAccountsForEmail(email);
+
+        for (Account account : accounts) {
+            accountsLogic.deleteAccount(account.getGoogleId());
+        }
+
+        deleteAccountCascadeAll(googleId);
```

Comment:
```suggestion
                if (usersDb.getAllUsersByGoogleId(googleId).isEmpty()) {
             accountsLogic.deleteAccountCascade(googleId);
         }
```

Please change accordingly for both.